cdap-app-fabric/src/main/java/co/cask/cdap/internal/provision/ProvisioningService.java

```diff
@@ -128,6 +146,55 @@ protected void shutDown() throws Exception {
     LOG.info("Stopped {}", getClass().getSimpleName());
   }
 
+  /**
+   * Scans the ProvisionerDataset for any tasks that should be in progress but are not being executed and consumes
+   * them.
+   */
+  @VisibleForTesting
+  void resumeTasks(Consumer<ProgramRunId> taskCleanup) {
+    // TODO: CDAP-13462 read tasks in chunks to avoid tx timeout
+    List<ProvisioningTaskInfo> clusterTaskInfos = getInProgressTasks();
+
+    for (ProvisioningTaskInfo provisioningTaskInfo : clusterTaskInfos) {
+      ProvisioningTaskKey taskKey = provisioningTaskInfo.getTaskKey();
```

Comment:
Probably should check if the service state is still running so that we can break the loop early if this service is getting shutdown.

Reply:
done